clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -224,10 +227,16 @@
                                                         + "Implementing the <code>ProducerInterceptor</code> interface allows you to intercept (and possibly mutate) the records "
                                                         + "received by the producer before they are published to the Kafka cluster. By default, there are no interceptors.";
 
+    /** <code>enable.idempotence</code> */
+    public static final String ENABLE_IDEMPOTENCE_CONFIG = "enable.idempotence";
+    public static final String ENABLE_IDEMPOTENCE_DOC = "When set to 'true', the producer will ensure that exactly one copy of each message is written in the stream. If 'false', producer "
```

Comment:
We should probably mention in the docs that enabling idempotence will change the default configurations. I think it might also be worth adding at least a `debug` level log message in the code that we are overriding the defaults.